src/common/common-thread-private.h

```diff
@@ -119,10 +119,14 @@ typedef struct {
  * libbson and libmongoc statically. */
 int
 mcommon_thread_join (bson_thread_t thread);
+// mcommon_thread_create returns 0 on success. Returns non-zero on error.
+// `errno_out` is set to an error code on error. Callers may use
+// `bson_strerror_r` to get an error message from `errno_out`.
 int
 mcommon_thread_create (bson_thread_t *thread,
                        BSON_THREAD_FUN_TYPE (func),
-                       void *arg);
+                       void *arg,
+                       int *errno_out);
```

Comment:
Could we simple return the error number as the return value of the function, rather than using the out-param?

Reply:
I like that idea. That simplifies the changes. Updated to return an error code rather than use an out-param.